src/ProductDefects.php

```diff
@@ -0,0 +1,173 @@
+
+/**
+ * @file
+ * Contains \Netzstrategen\ShopStandards\ProductDefects.
+ */
+
+namespace Netzstrategen\ShopStandards;
+
+/**
+ * Used or defective goods user consent checkbox functionality.
+ */
+class ProductDefects {
+
+  const FIELD_TITLE_TEXT = Plugin::PREFIX . '_product__defects_title_text';
+  const FIELD_DESCRIPTION_TEXT = Plugin::PREFIX . '_product_defects_description_text';
+  const FIELD_ATTRIBUTE_TEXT = Plugin::PREFIX . '_product_defects_attr_text';
+  const FIELD_PRODUCT_ATTRIBUTE_NAME = Plugin::L10N . '_product_defects_attribute_name';
+  const FIELD_PRODUCT_ATTRIBUTE_VALUE = Plugin::L10N . '_product_defects_attribute_value';
+
+  /**
+   * Checkbox consent initialization method.
+   */
+  public static function init() {
+    add_action('wp', __CLASS__ . '::display_product_checkbox');
+    add_filter('woocommerce_available_variation', __CLASS__ . '::pass_variation_attribute', 10, 3);
+    add_filter('woocommerce_get_settings_shop_standards', __CLASS__ . '::woocommerce_get_settings_shop_standards');
+  }
+
+  /**
+   * Determines if defective or used checkbox consent should be shown based on settings.
+   *
+   * @implemements wp
+   */
+  public static function display_product_checkbox() {
+    $product = wc_get_product();
+    if (!$product) {
+      return;
+    }
+
+    if (self::get_display_product_attribute($product)) {
+      add_action('woocommerce_before_add_to_cart_button', __CLASS__ . '::woocommerce_before_add_to_cart_button');
+      // Logs consent to defective or used product when product is added to cart.
+      add_action('woocommerce_add_to_cart', __CLASS__ . '::log_consent_file');
+    }
+  }
+
+  /**
+   * Returns the product attribute value to be displayed based on product and settings.
+   */
+  private static function get_display_product_attribute(\WC_Product $product): ?string {
+    $tax_name = get_option(self::FIELD_PRODUCT_ATTRIBUTE_NAME);
+    $tax_value = get_option(self::FIELD_PRODUCT_ATTRIBUTE_VALUE);
+    if (empty($tax_name) || empty($tax_value)) {
+      return NULL;
+    }
+
+    $attr_slug = 'pa_' . $tax_name;
+    $attr_value = $product->get_attribute($attr_slug);
+    if (empty($attr_value)) {
+      return NULL;
+    }
+    // Returns value when is not found in the product attrs as expected.
+    // But also when is a variable product which will be handled via JS.
+    return $product->is_type('variable') || strpos($attr_value, $tax_value) === FALSE ? $attr_value : NULL;
+  }
+
+  /**
+   * Logs consent to defective or used product.
+   *
+   * @implemements woocommerce_add_to_cart
+   */
+  public static function log_consent_file() {
+    if (!isset($_POST['used-goods-consent'])) {
+      return;
+    }
+
+    $uploads_dir = wp_upload_dir()['basedir'] . '/' . Plugin::PREFIX;
+    $log_file = $uploads_dir . '/defects-consent.log';
+    if (!dir($uploads_dir)) {
+      wp_mkdir_p($uploads_dir);
+    }
+    $data = [
+      'timestamp' => date_i18n('c'),
+      'user' => get_current_user_id(),
```

Comment:
Do we log this if the customer is not logged in?

Reply:
Frankly I just left this part pretty much untouched but I don't know the reason behind this, nothing written in the ticket and it was not mentioned by the PM's when discussed this, so I will better get rid of it